korman/nodes/node_avatar.py

```diff
@@ -71,3 +71,388 @@ def export(self, exporter, bo, so):
+            parent_MSB = self.find_output_socket("stage").links[0].to_node
+            idx = next((idx for idx, socket in enumerate(parent_MSB.find_input_sockets("stage_refs")) if socket.is_linked and socket.links[0].from_node == self))
+        return idx
+
+
+class PlasmaBehaviorSocket(PlasmaNodeSocketBase, bpy.types.NodeSocket):
+    bl_color = (0.348, 0.186, 0.349, 1.0)
+
+
+class PlasmaMultiStageBehaviorNode(PlasmaNodeBase, bpy.types.Node):
+    bl_category = "AVATAR"
+    bl_idname = "PlasmaMultiStageBehaviorNode"
+    bl_label = "Multistage Behavior"
+    bl_width_default = 200
+
+    pl_attrib = ("ptAttribBehavior")
+
+    freeze_phys = BoolProperty(name="Freeze Physical",
+                              description="Freeze physical at end",
+                              default=False)
+    reverse_control = BoolProperty(name="Reverse Controls",
+                              description="Reverse forward/back controls at end",
+                              default=False)
+
+    input_sockets = OrderedDict([
+        ("seek_target", {
+            "text": "Seek Target",
+            "type": "PlasmaSeekTargetSocketIn",
+            "valid_link_sockets": {"PlasmaSeekTargetSocketOut"},
+        }),
+        ("stage_refs", {
+            "text": "Stage",
+            "type": "PlasmaAnimStageRefSocket",
+            "valid_link_nodes": "PlasmaAnimStageNode",
+            "valid_link_sockets": "PlasmaAnimStageRefSocket",
+            "link_limit": 1,
+            "spawn_empty": True,
+        }),
+    ])
+
+    output_sockets = OrderedDict([
+        ("hosts", {
+            "text": "Host Script",
+            "type": "PlasmaBehaviorSocket",
+            "valid_link_nodes": {"PlasmaPythonFileNode"},
+            "spawn_empty": True,
+        })
+    ])
+
+    def draw_buttons(self, context, layout):
+        col = layout.column()
+        col.label("Options:")
+        col.prop(self, "freeze_phys")
+        col.prop(self, "reverse_control")
+
+    def get_key(self, exporter, so):
+        return self._find_create_key(plMultistageBehMod, exporter, so=so)
+
+    def export(self, exporter, bo, so):
+        seek_socket = self.find_input_socket("seek_target")
+
+        if seek_socket.is_linked:
+            smartSeek = True
+            seek_target = seek_socket.links[0].from_node.target
+        elif seek_socket.auto_target:
+            smartSeek = True
+            seek_target = bo
+        else:
+            smartSeek = False
+            seek_target = bo
+
+        # We need to make sure the object we're attached to is being exported as a Plasma Object, otherwise this is all nonsense
+        seek_target.plasma_object.enabled = True
+
+        if smartSeek:
+            spm = exporter.mgr.find_create_object(plSeekPointMod, name=seek_target.name, bl=seek_target)
+            spm.name = seek_target.name
+
+        msbmod = self._find_create_object(plMultistageBehMod, exporter, bl=seek_target)
```

Comment:
Since this is a single modifier, we need to be really careful about how we export it. This snippet will prevent the MSB from being exported more than once, duplicating the stages:
```python
    @property
    def export_once(self):
        return self.find_input_socket("seek_target").is_linked
```

Additionally, the pattern used by most nodes to get the "publlic" object is to call its own `get_key` method and grab the object from there. This is helpful for when code inevitable has to be tweaked :wink:

Reply:
I saw that in your recent PR, but wasn't sure if it was needed here.

Reply:
Why check if the seek_target is linked for this?  Shouldn't it check the `host` socket connection to the PFM?

Reply:
The seek target socket appears to be triggering whether or not we export a single MSB (seek point set) or multiple MSBs (seek point is the attached object).